seeds/node-nursery/src/vector-database.ts

```diff
@@ -14,17 +15,17 @@ import { similarity } from "ml-distance";
  *
  * Some databases, e.g. Pinecone, will also required a @field id field.
  */
-export interface VectorDocument {
+export interface VectorDocument extends Capability {
```

Comment:
That doesn't seem quite right. The VectorDatabase is a capability, but this and the query results are just plain data objects.

Did something error if you didn't do this?

BTW, the full typing should be creating a VectorDatabase<T> that stores VectorDocument<T> that returns VectorQueryResult<T>. What is interesting is that ideally we set <T> only once, probably where we add documents, and TypeScript should infer the type through all the connections.

Reply:
Added issue #97 to track this.